File: src/main/java/org/base/server/repository/ProjectRepository.java

package org.base.server.repository;

import java.util.Optional;

import org.base.server.entity.Project;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface ProjectRepository extends JpaRepository<Project, Long> {

    Optional<Project> findByProjectId(String projectId);

    Boolean existsByProjectId(String projectId);
}

File: src/main/kotlin/org/base/server/repository/ProjectRepository.kt
package org.base.server.repository

import org.base.server.entity.Project
import org.springframework.data.jpa.repository.JpaRepository
import java.util.Optional

interface ProjectRepository: JpaRepository<Project, Long> {
    fun findByProjectId(projectId: String): Optional<Project>
    fun existsByProjectId(projectId: String): Boolean
}